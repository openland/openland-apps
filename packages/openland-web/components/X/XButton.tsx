import * as React from 'react';
import Glamorous from 'glamorous';
import * as glamor from 'glamor';
import { XLink, XLinkProps } from './XLink';
import { withLayout, XLayoutProps } from './withLayout';
import { XIcon } from 'openland-x/XIcon';

function LoadingIcon(props: {white?: boolean}) {
    return (
        <svg className="loading-icon" xmlns="http://www.w3.org/2000/svg" xmlnsXlink="http://www.w3.org/1999/xlink" version="1.1" id="Capa_1" x="0px" y="0px" viewBox="0 0 512 512" style={{enableBackground: 'new 0 0 512 512'}} xmlSpace="preserve" width={512} height={512}>
            <g>
                <g>
                    <path d="M256.001,0c-8.284,0-15,6.716-15,15v96.4c0,8.284,6.716,15,15,15s15-6.716,15-15V15C271.001,6.716,264.285,0,256.001,0z" fill={props.white ? '#fff' : '#933EC5'} />
                </g>
            </g>
            <g>
                <g>
                    <path d="M256.001,385.601c-8.284,0-15,6.716-15,15V497c0,8.284,6.716,15,15,15s15-6.716,15-15v-96.399    C271.001,392.316,264.285,385.601,256.001,385.601z" fill={props.white ? '#fff' : '#933EC5'} />
                </g>
            </g>
            <g>
                <g>
                    <path d="M196.691,123.272l-48.2-83.485c-4.142-7.175-13.316-9.633-20.49-5.49c-7.174,4.142-9.632,13.316-5.49,20.49l48.2,83.485    c2.778,4.813,7.82,7.502,13.004,7.502c2.545,0,5.124-0.648,7.486-2.012C198.375,139.62,200.833,130.446,196.691,123.272z" fill={props.white ? '#fff' : '#933EC5'} />
                </g>
            </g>
            <g>
                <g>
                    <path d="M389.491,457.212l-48.199-83.483c-4.142-7.175-13.316-9.633-20.49-5.49c-7.174,4.142-9.632,13.316-5.49,20.49    l48.199,83.483c2.778,4.813,7.82,7.502,13.004,7.502c2.545,0,5.124-0.648,7.486-2.012    C391.175,473.56,393.633,464.386,389.491,457.212z" fill={props.white ? '#fff' : '#933EC5'} />
                </g>
            </g>
            <g>
                <g>
                    <path d="M138.274,170.711L54.788,122.51c-7.176-4.144-16.348-1.685-20.49,5.49c-4.142,7.174-1.684,16.348,5.49,20.49    l83.486,48.202c2.362,1.364,4.941,2.012,7.486,2.012c5.184,0,10.226-2.69,13.004-7.503    C147.906,184.027,145.448,174.853,138.274,170.711z" fill={props.white ? '#fff' : '#933EC5'} />
                </g>
            </g>
            <g>
                <g>
                    <path d="M472.213,363.51l-83.484-48.199c-7.176-4.142-16.349-1.684-20.49,5.491c-4.142,7.175-1.684,16.349,5.49,20.49    l83.484,48.199c2.363,1.364,4.941,2.012,7.486,2.012c5.184,0,10.227-2.69,13.004-7.502    C481.845,376.825,479.387,367.651,472.213,363.51z" fill={props.white ? '#fff' : '#933EC5'} />
                </g>
            </g>
            <g>
                <g>
                    <path d="M111.401,241.002H15c-8.284,0-15,6.716-15,15s6.716,15,15,15h96.401c8.284,0,15-6.716,15-15    S119.685,241.002,111.401,241.002z" fill={props.white ? '#fff' : '#933EC5'} />
                </g>
            </g>
            <g>
                <g>
                    <path d="M497,241.002h-96.398c-8.284,0-15,6.716-15,15s6.716,15,15,15H497c8.284,0,15-6.716,15-15S505.284,241.002,497,241.002z" fill={props.white ? '#fff' : '#933EC5'} />
                </g>
            </g>
            <g>
                <g>
                    <path d="M143.765,320.802c-4.142-7.175-13.314-9.633-20.49-5.49l-83.486,48.2c-7.174,4.142-9.632,13.316-5.49,20.49    c2.778,4.813,7.82,7.502,13.004,7.502c2.545,0,5.124-0.648,7.486-2.012l83.486-48.2    C145.449,337.15,147.907,327.976,143.765,320.802z" fill={props.white ? '#fff' : '#933EC5'} />
                </g>
            </g>
            <g>
                <g>
                    <path d="M477.702,128.003c-4.142-7.175-13.315-9.632-20.49-5.49l-83.484,48.2c-7.174,4.141-9.632,13.315-5.49,20.489    c2.778,4.813,7.82,7.503,13.004,7.503c2.544,0,5.124-0.648,7.486-2.012l83.484-48.2    C479.386,144.351,481.844,135.177,477.702,128.003z" fill={props.white ? '#fff' : '#933EC5'} />
                </g>
            </g>
            <g>
                <g>
                    <path d="M191.201,368.239c-7.174-4.144-16.349-1.685-20.49,5.49l-48.2,83.485c-4.142,7.174-1.684,16.348,5.49,20.49    c2.362,1.364,4.941,2.012,7.486,2.012c5.184,0,10.227-2.69,13.004-7.502l48.2-83.485    C200.833,381.555,198.375,372.381,191.201,368.239z" fill={props.white ? '#fff' : '#933EC5'} />
                </g>
            </g>
            <g>
                <g>
                    <path d="M384.001,34.3c-7.175-4.144-16.349-1.685-20.49,5.49l-48.199,83.483c-4.143,7.174-1.685,16.348,5.49,20.49    c2.362,1.364,4.941,2.012,7.486,2.012c5.184,0,10.226-2.69,13.004-7.502l48.199-83.483    C393.633,47.616,391.175,38.442,384.001,34.3z" fill={props.white ? '#fff' : '#933EC5'} />
                </g>
            </g>
        </svg>
    );
}

export interface XButtonStyleProps extends XLayoutProps {
    style?: 'normal' | 'dark' | 'important' | 'google';
    size?: 'large' | 'medium' | 'normal';
    bounce?: boolean;
    loading?: boolean;
    disabled?: boolean;
    icon?: string;
    accent?: boolean;
    borderless?: boolean;
}

export interface XButtonProps extends XLinkProps, XButtonStyleProps {

}

const loading = glamor.keyframes({
    '0%': { transform: `rotate(0deg)` },
    '100%': { transform: `rotate(360deg)` }
});

let textColors = {
    'normal': '#525f7f',
    'dark': '#ffffff',
    'important': '#ffffff',
    'google': '#525f7f'
};

let textHoveredColors = {
    'normal': '#32325d',
    'dark': '#ffffff',
    'important': '#ffffff',
    'google': '#32325d'
};

let backgroundColors = {
    'normal': '#ffffff',
    'dark': '#6B50FF',
    'important': '#3ecf8e',
    'google': '#ffffff'
};

let backgroundHoveredColors = {
    'normal': undefined,
    'dark': '#8571f3',
    'important': undefined,
    'google': undefined
};

let backgroundPressedColors = {
    'normal': undefined,
    'dark': '#5032f3',
    'important': undefined,
    'google': undefined
};

let fontSize = {
    'normal': '13px',
    'medium': '14px',
    'large': '15px'
};

let iconSize = {
    'normal': '15px',
    'medium': '18px',
    'large': '20px'
};

let paddings = {
    'normal': '6px 14px',
    'medium': '10px 18px',
    'large': '16px 20px'
};

export const XButtonComponent = withLayout(Glamorous<XButtonProps>(XLink)((props) => {
    let style = props.style !== undefined && props.style !== 'normal' ? props.style : 'normal';
    let size = props.size !== undefined && props.size !== 'normal' ? props.size : 'normal';
    return {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        textDecoration: 'none',
        textAlign: props.icon ? 'right' : 'center',
        cursor: (props.loading || props.disabled) ? 'inherit' : 'pointer',
        userSelect: 'none',
        whiteSpace: 'nowrap',
        wordBreak: 'keep-all',
        position: 'relative',
        minHeight: '32px',
        minWidth: '32px',
        outline: 'none',
        opacity: props.disabled ? 0.8 : 1,
        pointerEvents: (props.loading || props.disabled) ? 'none' : 'auto',

        padding: props.borderless ? undefined : paddings[size],

        color: props.accent ? '#4428e0' : (props.loading ? 'transparent' : textColors[style]),
        backgroundColor: props.accent ? '#fff' : backgroundColors[style],

        border: props.borderless ? undefined : (props.accent ? '1px solid #4428e0' : undefined),
        borderRadius: '4px',

        boxShadow: props.borderless ? undefined : ('0 0 0 1px rgba(50,50,93,.1), 0 2px 5px 0 rgba(50,50,93,.08), 0 1px 1.5px 0 rgba(0,0,0,.07), 0 1px 2px 0 rgba(0,0,0,.08), 0 0 0 0 transparent'),
        transition: 'box-shadow .08s ease-in,color .08s ease-in,all .15s ease',

        fontSize: fontSize[size],
        lineHeight: '20px',
        fontWeight: 500,

        '&:hover': {
            transform: props.bounce ? 'translateY(-1px)' : undefined,
            color: props.accent ? '#4428e0' : (props.loading ? 'transparent' : textHoveredColors[style]),
            backgroundColor: (props.loading || props.disabled) ? backgroundColors[style] : backgroundHoveredColors[style],
            boxShadow: props.borderless ? undefined : (((props.loading || props.disabled) ? undefined
                : ((props.size === 'large')
                    ? '0 7px 14px rgba(50,50,93,.1), 0 3px 6px rgba(0,0,0,.08)'
                    : '0 0 0 1px rgba(50,50,93,.1), 0 2px 5px 0 rgba(50,50,93,.1), 0 3px 9px 0 rgba(50,50,93,.08), 0 1px 1.5px 0 rgba(0,0,0,.08), 0 1px 2px 0 rgba(0,0,0,.08)'
                )
            ))
        },
        '&:active': {
            transform: props.bounce ? 'translateY(1px)' : undefined,
            color: props.loading ? 'transparent' : textHoveredColors[style],
            backgroundColor: (props.loading || props.disabled) ? backgroundColors[style] : backgroundPressedColors[style],
            boxShadow: props.borderless ? undefined : (((props.loading || props.disabled) ? undefined
                : ((props.size === 'large')
                    ? '0 4px 6px rgba(50,50,93,.11), 0 1px 3px rgba(0,0,0,.08)'
                    : '0 0 0 1px rgba(50,50,93,.08), 0 2px 5px 0 rgba(50,50,93,.06), 0 1px 1.5px 0 rgba(0,0,0,.05), 0 1px 2px 0 rgba(0,0,0,.06), 0 0 0 0 transparent'
                )
            ))
        },
        '&:focus': {
            boxShadow: (props.loading || props.disabled || props.borderless)
                ? undefined
                : '0 0 0 1px rgba(50,151,211,.2), 0 0 0 2px rgba(50,151,211,.25), 0 2px 5px 0 rgba(0,0,0,.1), 0 0 0 0 transparent, 0 0 0 0 transparent',
        },
        '& > i': {
            opacity: props.loading ? 0 : 1,
            fontSize: iconSize[size],
            lineHeight: '20px'
        },
        '& > span': {
            marginLeft: props.icon ? (props.accent ? 7 : 3) : 0
        },
        '& > .loading-icon': {
            display: 'block',
            position: 'absolute',
            width: '20px',
            height: '20px',
            left: 'calc(50% - 10px)',
            top: 'calc(50% - 10px)',
            animation: `${loading} 2s linear infinite`
        }
    };
}));

export function XButton(props: XButtonProps & { children?: any }) {
    let { icon, children, ...other } = props;
    return (
        <XButtonComponent {...other} icon={icon}>
            {icon && <XIcon icon={icon} />}
            {children && (<span>{children}</span>)}
            {props.loading && <LoadingIcon white={(props.style === 'dark' || props.style === 'important') ? true : false}/>}
        </XButtonComponent>
    );
}