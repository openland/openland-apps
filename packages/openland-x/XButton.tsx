import * as React from 'react';
import Glamorous from 'glamorous';
import { styleResolver } from 'openland-x-utils/styleResolver';
import { XLink, XLinkProps } from './XLink';
import { XLoadingCircular } from './XLoadingCircular';
import { XFlexStyles, applyFlex } from './Flex';
import { XIcon } from './XIcon';

export interface XButtonStyleProps extends XFlexStyles {
    text?: string;
    icon?: string;
    size?: 'x-large' | 'large' | 'medium' | 'default' | 'small';
    style?: 'primary' | 'danger' | 'default' | 'ghost' | 'electric' | 'flat';
}

export interface XButtonProps extends XButtonStyleProps, XLinkProps {
    loading?: boolean;
}

let iconsIndentation = styleResolver({
    'x-large': {
        marginLeft: -8,
        marginRight: 12
    },
    'large': {
        marginRight: 10
    },
    'medium': {
        marginRight: 8
    },
    'default': {
        marginRight: 6
    },
    'small': {
        marginRight: 5
    }
});

let sizeStyles = styleResolver({
    'x-large': {
        height: 56,
        lineHeight: '56px',
        fontSize: 18,
        letterSpacing: 0.6,
        fontWeight: 500,
        borderRadius: 4,
        '& .button-content': {
            paddingLeft: 32,
            paddingRight: 32
        },
        '& .icon': {
            width: 32,
            fontSize: 28,
            lineHeight: '56px',
        },
        '& .loading-icon': {
            width: 24,
            height: 24,
            lineHeight: 'normal',
            top: 'calc(50% - 12px)',
            left: 'calc(50% - 12px)'
        }
    },
    'large': {
        height: 48,
        lineHeight: '48px',
        fontSize: 18,
        letterSpacing: 0.6,
        fontWeight: 500,
        borderRadius: 4,
        '& .button-content': {
            paddingLeft: 26,
            paddingRight: 26
        },
        '& .icon': {
            fontSize: 24
        },
        '& .loading-icon': {
            width: 21,
            height: 21,
            lineHeight: 'normal',
            top: 'calc(50% - 10.5px)',
            left: 'calc(50% - 10.5px)'
        }
    },
    'medium': {
        height: 40,
        lineHeight: '40px',
        fontSize: 16,
        letterSpacing: 0.5,
        fontWeight: 500,
        borderRadius: 4,
        '& .button-content': {
            paddingLeft: 20,
            paddingRight: 20
        },
        '& .icon': {
            fontSize: 20
        },
        '& .loading-icon': {
            width: 18,
            height: 18,
            lineHeight: 'normal',
            top: 'calc(50% - 9px)',
            left: 'calc(50% - 9px)'
        }
    },
    'default': {
        height: 32,
        lineHeight: '32px',
        fontSize: 14,
        letterSpacing: 0.4,
        fontWeight: 500,
        borderRadius: 4,
        '& .button-content': {
            paddingLeft: 14,
            paddingRight: 14
        },
        '& .icon': {
            fontSize: 16
        },
        '& .loading-icon': {
            width: 15,
            height: 15,
            lineHeight: 'normal',
            top: 'calc(50% - 7.5px)',
            left: 'calc(50% - 7.5px)'
        }
    },
    'small': {
        height: 24,
        lineHeight: '24px',
        fontSize: 12,
        letterSpacing: 0.4,
        fontWeight: 500,
        borderRadius: 3,
        '& .button-content': {
            paddingLeft: 10,
            paddingRight: 10
        },
        '& .icon': {
            fontSize: 14
        },
        '& .loading-icon': {
            width: 14,
            height: 14,
            lineHeight: 'normal',
            top: 'calc(50% - 7px)',
            left: 'calc(50% - 7px)'
        }
    }
});

let colorStyles = styleResolver({
    'default': {
        backgroundColor: '#f3f3f5',
        color: '#334562',
        '&:hover': {
            backgroundColor: '#ecedf0',
            color: '#334562'
        },
        '&:active': {
            backgroundColor: '#eeecfa',
            color: '#5640d6'
        },
        '&:focus': {
            boxShadow: '0 0 0 1px rgba(50,151,211,.2), 0 0 0 2px rgba(50,151,211,.25), 0 2px 5px 0 rgba(0,0,0,.1), 0 0 0 0 transparent, 0 0 0 0 transparent',
        }
    },
    'primary': {
        backgroundColor: '#654bfa',
        color: '#ffffff',
        '&:hover': {
            backgroundColor: '#7159f9',
            color: '#ffffff'
        },
        '&:active': {
            backgroundColor: '#5640d6',
            color: '#ffffff'
        },
        '&:focus': {
            boxShadow: '0 0 0 1px rgba(50,151,211,.2), 0 0 0 2px rgba(50,151,211,.25), 0 2px 5px 0 rgba(0,0,0,.1), 0 0 0 0 transparent, 0 0 0 0 transparent',
        }
    },
    'danger': {
        backgroundColor: '#d75454',
        color: '#ffffff',
        '&:hover': {
            backgroundColor: '#ec6262',
            color: '#ffffff'
        },
        '&:active': {
            backgroundColor: '#c54f4f',
            color: '#ffffff'
        },
        '&:focus': {
            boxShadow: '0 0 0 1px rgba(50,151,211,.2), 0 0 0 2px rgba(50,151,211,.25), 0 2px 5px 0 rgba(0,0,0,.1), 0 0 0 0 transparent, 0 0 0 0 transparent',
        }
    },
    'ghost': {
        backgroundColor: '#ffffff',
        color: '#1f3449',
        border: 'solid 1px #c2cbde',
        '&:hover': {
            backgroundColor: '#f3f3f5',
            color: '#334562'
        },
        '&:active': {
            backgroundColor: '#eeecfa',
            color: '#5640d6'
        },
        '&:focus': {
            boxShadow: '0 0 0 1px rgba(50,151,211,.2), 0 0 0 2px rgba(50,151,211,.25), 0 2px 5px 0 rgba(0,0,0,.1), 0 0 0 0 transparent, 0 0 0 0 transparent',
        }
    },
    'electric': {
        backgroundColor: '#ffffff',
        color: '#5640d6',
        border: 'solid 1px #654bfa',
        '&:hover': {
            backgroundColor: '#eeecfa',
        },
        '&:active': {
            backgroundColor: '#eeecfa',
            border: 'solid 1px transparent',
        },
        '&:focus': {
            boxShadow: '0 0 0 1px rgba(50,151,211,.2), 0 0 0 2px rgba(50,151,211,.25), 0 2px 5px 0 rgba(0,0,0,.1), 0 0 0 0 transparent, 0 0 0 0 transparent',
        }
    },
    'flat': {
        backgroundColor: '#ffffff',
        color: '#334562',
        border: 'solid 1px transparent',
        '&:hover': {
            color: '#334562',
            backgroundColor: '#f4f6fb',
            border: 'solid 1px #c2cbde'
        },
        '&:active': {
            backgroundColor: '#eeecfa',
            border: 'solid 1px transparent',
            color: '#5640d6'
        },
        '&:focus': {
            boxShadow: '0 0 0 1px rgba(50,151,211,.2), 0 0 0 2px rgba(50,151,211,.25), 0 2px 5px 0 rgba(0,0,0,.1), 0 0 0 0 transparent, 0 0 0 0 transparent',
        }
    }
});

let colorDisabledStyles = styleResolver({
    'default': {
        backgroundColor: '#f3f3f5 !important',
        color: 'rgba(51, 69, 98, 0.7) !important',
    },
    'primary': {
        backgroundColor: '#9380fc !important',
        color: 'rgba(255, 255, 255, 0.7) !important',
    },
    'danger': {
        backgroundColor: '#e28787 !important',
        color: 'rgba(255, 255, 255, 0.7) !important',
    },
    'ghost': {
        backgroundColor: '#ffffff !important',
        color: 'rgba(51, 69, 98, 0.7) !important',
        border: 'solid 1px #c2cbde !important',
    },
    'electric': {
        backgroundColor: '#ffffff !important',
        color: 'rgba(86, 64, 214, 0.7) !important',
        border: 'solid 1px #b1a4f9 !important',
    },
    'flat': {
        backgroundColor: '#ffffff !important',
        color: 'rgba(51, 69, 98, 0.7) !important',
    }
});

let loaderStyles = styleResolver({
    'default': {
        '& svg path': {
            stroke: '#334562 !important'
        }
    },
    'primary': {
        '& svg path': {
            stroke: '#ffffff !important'
        }
    },
    'danger': {
        '& svg path': {
            stroke: '#ffffff !important'
        }
    },
    'ghost': {
        '& svg path': {
            stroke: '#334562 !important'
        }
    },
    'electric': {
        '& svg path': {
            stroke: '#5640d6 !important'
        }
    },
    'flat': {
        '& svg path': {
            stroke: '#334562 !important'
        }
    }
});

const StyledIcon = Glamorous<XButtonProps>(XIcon)([
    (props) => iconsIndentation(props.size, !!props.text)
]);

const StyledButtonContentWrapper = Glamorous.div({
    width: '100%',
    height: '100%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    textDecoration: 'none',
    flexDirection: 'row',
    userSelect: 'none',
    whiteSpace: 'nowrap',
    wordBreak: 'keep-all',
    position: 'relative',
    outline: 'none'
});

const StyledButton = Glamorous<XButtonProps>(XLink)([
    (props) => ({
        pointerEvents: (props.loading || props.disabled) ? 'none' : 'auto',
        cursor: (props.loading || props.disabled) ? 'inherit' : 'pointer',
        transition: 'box-shadow .08s ease-in,color .08s ease-in, border .0s, all .15s ease'
    }),
    (props) => (props.loading && {
        '& span': { opacity: 0 }
    } || {}),
    (props) => colorStyles(props.style, !props.disabled),
    (props) => colorDisabledStyles(props.style, !!props.disabled),
    (props) => loaderStyles(props.style),
    (props) => sizeStyles(props.size),
    (props) => applyFlex(props)
]);

export class XButton extends React.PureComponent<XButtonProps> {
    render() {
        return (
            <StyledButton {...this.props}>
                <StyledButtonContentWrapper tabIndex={-1} className="button-content">
                    {this.props.icon && <StyledIcon text={this.props.text} icon={this.props.icon} className="icon" />}
                    <span>{this.props.text}</span>
                    {this.props.loading && <XLoadingCircular inverted={this.props.style === 'primary' || this.props.style === 'danger'} className="loading-icon" />}
                </StyledButtonContentWrapper>
            </StyledButton>
        );
    }
}