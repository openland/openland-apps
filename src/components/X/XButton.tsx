import * as glamor from 'glamor'
import { XLink, XLinkProps } from './XLink';
import XStyled from './XStyled';

interface XButtonProps extends XLinkProps {
    alignSelf?: 'stretch' | 'flex-start' | 'flex-end' | 'center';
    style?: 'normal' | 'dark' | 'important';
    size?: 'large' | 'normal';
    bounce?: boolean;
    loading?: boolean;
}

const loading = glamor.keyframes({
    '0%': { transform: `rotate(0deg)` },
    '100%': { transform: `rotate(360deg)` }
})

let textColors = {
    'normal': '#525f7f',
    'dark': '#ffffff',
    'important': '#ffffff'
}
let textHoveredColors = {
    'normal': '#32325d',
    'dark': '#ffffff',
    'important': '#ffffff'
}
let backgroundColors = {
    'normal': '#ffffff',
    'dark': '#6B50FF',
    'important': '#3ecf8e'
}
let backgroundHoveredColors = {
    'normal': undefined,
    'dark': '#8571f3',
    'important': undefined
}
let backgroundPressedColors = {
    'normal': undefined,
    'dark': '#5032f3',
    'important': undefined
}

export const XButton = XStyled<XButtonProps>(XLink)((props) => {
    let style = props.style !== undefined && props.style !== 'normal' ? props.style : 'normal'
    return {
        textDecoration: 'none',
        textAlign: 'center',
        cursor: props.loading ? 'inherit' : 'pointer',
        userSelect: 'none',
        whiteSpace: 'nowrap',
        wordBreak: 'keep-all',
        position: 'relative',

        padding: props.size === 'large' ? '16px 20px' : '6px 14px',

        color: textColors[style],
        backgroundColor: backgroundColors[style],

        borderRadius: '4px',

        boxShadow: '0 0 0 1px rgba(50,50,93,.1), 0 2px 5px 0 rgba(50,50,93,.08), 0 1px 1.5px 0 rgba(0,0,0,.07), 0 1px 2px 0 rgba(0,0,0,.08), 0 0 0 0 transparent',
        transition: 'box-shadow .08s ease-in,color .08s ease-in,all .15s ease',

        fontSize: props.size === 'large' ? '15px' : '13px',
        lineHeight: '20px',
        fontWeight: 500,

        alignSelf: props.alignSelf,

        '&:hover': {
            transform: props.bounce ? 'translateY(-1px)' : undefined,
            color: textHoveredColors[style],
            backgroundColor: backgroundHoveredColors[style],
            boxShadow: props.size === 'large'
                ? '0 7px 14px rgba(50,50,93,.1), 0 3px 6px rgba(0,0,0,.08)'
                : '0 0 0 1px rgba(50,50,93,.1), 0 2px 5px 0 rgba(50,50,93,.1), 0 3px 9px 0 rgba(50,50,93,.08), 0 1px 1.5px 0 rgba(0,0,0,.08), 0 1px 2px 0 rgba(0,0,0,.08)'
        },
        '&:active': {
            transform: props.bounce ? 'translateY(1px)' : undefined,
            color: textHoveredColors[style],
            backgroundColor: backgroundPressedColors[style],
            boxShadow: props.size === 'large'
                ? '0 4px 6px rgba(50,50,93,.11), 0 1px 3px rgba(0,0,0,.08)'
                : '0 0 0 1px rgba(50,50,93,.08), 0 2px 5px 0 rgba(50,50,93,.06), 0 1px 1.5px 0 rgba(0,0,0,.05), 0 1px 2px 0 rgba(0,0,0,.06), 0 0 0 0 transparent'
        },
        '&:focus': {
            boxShadow: '0 0 0 1px rgba(50,151,211,.2), 0 0 0 2px rgba(50,151,211,.25), 0 2px 5px 0 rgba(0,0,0,.1), 0 0 0 0 transparent, 0 0 0 0 transparent',
            outline: 0
        },
        '&::after': {
            content: props.loading ? `''` : undefined,
            display: 'block',
            position: 'absolute',
            width: '20px',
            height: '20px',
            left: 'calc(50% - 10px)',
            top: 'calc(50% - 10px)',
            backgroundImage: 'url(/static/X/loading.svg)',
            backgroundSize: '20px',
            animation: `${loading} 2s linear infinite`
        }
    }
});