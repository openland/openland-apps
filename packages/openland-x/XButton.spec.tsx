import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { XButton } from './XButton';

describe('XButton', () => {
    it('should render correctly', () => {
        expect(renderer.create(<XButton text="X Button Test" />).toJSON()).toMatchSnapshot();
    });
});