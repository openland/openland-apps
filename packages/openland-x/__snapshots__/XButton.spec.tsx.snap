// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XButton should render correctly 1`] = `
.css-126rrgb,
[data-css-126rrgb] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  box-sizing: border-box;
  pointer-events: auto;
  cursor: pointer;
  transition: box-shadow 0.08s ease-in, color 0.08s ease-in, border 0s,
    all 0.15s ease;
  background-color: #f6f6f6;
  color: rgba(0, 0, 0, 0.7);
  border: solid 1px transparent;
  height: 32px;
  line-height: 30px;
  font-size: 13px;
  letter-spacing: 0;
  font-weight: 600;
  border-top-left-radius: 16px;
  border-bottom-left-radius: 16px;
  border-top-right-radius: 16px;
  border-bottom-right-radius: 16px;
  -webkit-transition: box-shadow 0.08s ease-in, color 0.08s ease-in, border 0s,
    all 0.15s ease;
  -moz-transition: box-shadow 0.08s ease-in, color 0.08s ease-in, border 0s,
    all 0.15s ease;
}
.css-126rrgb .loading-icon,
[data-css-126rrgb] .loading-icon {
  position: absolute;
  width: 32px;
  height: 32px;
  line-height: normal;
  top: calc(50% - 16px);
  left: calc(50% - 16px);
}
.css-126rrgb .icon-responsive,
[data-css-126rrgb] .icon-responsive {
  display: none;
}
.css-126rrgb:hover,
[data-css-126rrgb]:hover,
.css-126rrgb:hover *,
[data-css-126rrgb]:hover * {
  text-decoration: none !important;
}
.css-126rrgb:hover,
[data-css-126rrgb]:hover {
  background-color: #e6e9ec;
  color: rgba(0, 0, 0, 0.7);
}
.css-126rrgb:active,
[data-css-126rrgb]:active {
  background-color: #1790ff;
  color: #ffffff;
}
.css-126rrgb:focus,
[data-css-126rrgb]:focus {
  box-shadow: 0 0 0 1px rgba(50, 151, 211, 0.2),
    0 0 0 2px rgba(50, 151, 211, 0.25), 0 2px 5px 0 rgba(0, 0, 0, 0.1),
    0 0 0 0 transparent, 0 0 0 0 transparent;
}
.css-126rrgb .button-content,
[data-css-126rrgb] .button-content {
  padding-left: 16px;
  padding-right: 16px;
}
.css-126rrgb .icon.material,
[data-css-126rrgb] .icon.material {
  font-size: 20px;
}
.css-126rrgb i.icon-svg,
[data-css-126rrgb] i.icon-svg {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
}
.css-126rrgb svg,
[data-css-126rrgb] svg {
  margin-left: -2px;
  margin-right: 4px;
}
.css-11a8r8c,
[data-css-11a8r8c] {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  text-decoration: none;
  flex-direction: row;
  user-select: none;
  white-space: nowrap;
  word-break: keep-all;
  position: relative;
  outline: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}
.css-12alag6,
[data-css-12alag6] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  align-items: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}
.css-n8ddkv,
[data-css-n8ddkv] {
  max-width: 100%;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.css-0,
[data-css-0] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  box-sizing: border-box;
  pointer-events: auto;
  cursor: pointer;
  transition: box-shadow .08s ease-in,color .08s ease-in, border .0s, all .15s ease;
  background-color: #f6f6f6;
  color: rgba(0, 0, 0, 0.7);
  border: solid 1px transparent;
  height: 32px;
  line-height: 30px;
  font-size: 13px;
  letter-spacing: 0;
  font-weight: 600;
  border-top-left-radius: 16px;
  border-bottom-left-radius: 16px;
  border-top-right-radius: 16px;
  border-bottom-right-radius: 16px;
  -webkit-transition: box-shadow .08s ease-in,color .08s ease-in, border .0s, all .15s ease;
  -moz-transition: box-shadow .08s ease-in,color .08s ease-in, border .0s, all .15s ease;
}

.css-0 .loading-icon,
[data-css-0] .loading-icon {
  position: absolute;
  width: 32px;
  height: 32px;
  line-height: normal;
  top: calc(50% - 16px);
  left: calc(50% - 16px);
}

.css-0 .icon-responsive,
[data-css-0] .icon-responsive {
  display: none;
}

.css-0:hover,
[data-css-0]:hover,
.css-0:hover *,
[data-css-0]:hover * {
  text-decoration: none !important;
}

.css-0:hover,
[data-css-0]:hover {
  background-color: #e6e9ec;
  color: rgba(0, 0, 0, 0.7);
}

.css-0:active,
[data-css-0]:active {
  background-color: #1790ff;
  color: #ffffff;
}

.css-0:focus,
[data-css-0]:focus {
  box-shadow: 0 0 0 1px rgba(50,151,211,.2), 0 0 0 2px rgba(50,151,211,.25), 0 2px 5px 0 rgba(0,0,0,.1), 0 0 0 0 transparent, 0 0 0 0 transparent;
}

.css-0 .button-content,
[data-css-0] .button-content {
  padding-left: 16px;
  padding-right: 16px;
}

.css-0 .icon.material,
[data-css-0] .icon.material {
  font-size: 20px;
}

.css-0 i.icon-svg,
[data-css-0] i.icon-svg {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
}

.css-0 svg,
[data-css-0] svg {
  margin-left: -2px;
  margin-right: 4px;
}

.css-4,
[data-css-4] {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  justify-content: center;
  align-items: center;
  text-decoration: none;
  flex-direction: row;
  user-select: none;
  white-space: nowrap;
  word-break: keep-all;
  position: relative;
  outline: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

.css-5,
[data-css-5] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  align-items: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-6,
[data-css-6] {
  max-width: 100%;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

<a
  className="css-0"
>
  <div
    className="css-4 button-content"
    tabIndex={-1}
  >
    <div
      className="css-5 main-content"
    >
      <span
        className="css-6"
      >
        X Button Test
      </span>
    </div>
  </div>
</a>
`;
